### PR TITLE
[DOC] Document TTreeReaderValue::Get

### PR DESCRIPTION
**Pull request by bluehood**

### CODE REVIEW

File: tree/treeplayer/inc/TTreeReaderValue.h [hunk 1]
```diff
@@ -132,14 +132,22 @@ class TTreeReaderValue: public ROOT::Internal::TTreeReaderValueBase {
       TTreeReaderValueBase(&tr, branchname,
                            TDictionary::GetDictionary(typeid(NonConstT_t))) {}
 
+   /// Return a pointer to the value of the current entry.
+   /// Return a nullptr and print an error if no entry has been loaded yet.
+   /// The returned address is guaranteed to stay constant while a given TTree is being read from a given file.
```
Comment (Axel-Naumann) [inline tree/treeplayer/inc/TTreeReaderValue.h]: ...unless the branch addresses are manipulated directly (e.g. through `TTree::SetBranchAddress()`).

File: tree/treeplayer/inc/TTreeReaderValue.h [hunk 2]
```diff
@@ -132,14 +132,22 @@ class TTreeReaderValue: public ROOT::Internal::TTreeReaderValueBase {
...
-         return 0;
+         return nullptr;
       }
       void *address = GetAddress(); // Needed to figure out if it's a pointer
       return fProxy->IsaPointer() ? *(T**)address : (T*)address; }
+   /// Return a pointer to the value of the current entry.
+   /// Equivalent to Get.
    T* operator->() { return Get(); }
+   /// Return a reference to the value of the current entry.
+   /// Equivalent to dereferencing the pointer returned by Get. Behavior is undefined if no entry has been loaded yet.
```
Comment (Axel-Naumann) [inline tree/treeplayer/inc/TTreeReaderValue.h]: "Behavior is undefined" is the correct technical term - "will likely crash" is the more user-oriented one :-) I don't have a strong preference here... in principle this should throw if `Get()` returns `nullptr`.

Comment (Axel-Naumann) [inline tree/treeplayer/inc/TTreeReaderValue.h]: Get -> `Get()`?

File: tree/treeplayer/inc/TTreeReaderValue.h [hunk 3]
```diff
@@ -132,14 +132,22 @@ class TTreeReaderValue: public ROOT::Internal::TTreeReaderValueBase {
...
    T* Get() {
       if (!fProxy){
          Error("TTreeReaderValue::Get()", "Value reader not properly initialized, did you remember to call TTreeReader.Set(Next)Entry()?");
-         return 0;
+         return nullptr;
       }
       void *address = GetAddress(); // Needed to figure out if it's a pointer
       return fProxy->IsaPointer() ? *(T**)address : (T*)address; }
+   /// Return a pointer to the value of the current entry.
+   /// Equivalent to Get.
```
Comment (Axel-Naumann) [inline tree/treeplayer/inc/TTreeReaderValue.h]: Get -> `Get()`?

Comment (bluehood) [reply tree/treeplayer/inc/TTreeReaderValue.h]: I hoped Doxygen was smart enough to highlight that and add an anchor link to the `Get` method I mention. Not sure if that's the case, I will test it locally.

File: tree/treeplayer/inc/TTreeReaderValue.h (hunk 1 above)
Comment (bluehood) [reply tree/treeplayer/inc/TTreeReaderValue.h]: will add this to the text

File: tree/treeplayer/inc/TTreeReaderValue.h (hunk 2 above)
Comment (bluehood) [reply tree/treeplayer/inc/TTreeReaderValue.h]: will add a note about the crashing :)

Comment (bluehood) [reply tree/treeplayer/inc/TTreeReaderValue.h]: see above

**Comment by bluehood** [comment]
All comments have been addressed.
I would love a way to tell jenkins it doesn't need to run this :D

**Comment by bluehood** [comment]
hi @Axel-Naumann , can I merge?

**Pull request merged**